pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiEventConfigurationOccurrenceCounter.java

```diff
@@ -37,19 +36,34 @@ default Integer getOrCreateNumOccurrences(long eventConfigurationId, long partic
         );
     }
 
+    default void incNumOccurrences(long eventConfigurationId, long participantUserId) {
+        _getNumOccurrencesForUpdate(eventConfigurationId, participantUserId)
+                .ifPresent(previousCounterValue ->
+                        _updateNumOccurrences(eventConfigurationId, participantUserId, previousCounterValue + 1));
+    }
+
+    // private
+    @SqlQuery(NUM_OCCURRENCES_SELECT_SQL + " FOR UPDATE")
+    Optional<Integer> _getNumOccurrencesForUpdate(
+            @Bind("eventConfigurationId") long eventConfigurationId,
+            @Bind("participantUserId") long participantUserId
+    );
+
+    // private
     @SqlUpdate(
-            "update event_configuration_occurrence_counter set num_occurrences = num_occurrences + 1"
```

Comment:
I will be surprised if mysql can't assure that counter = counter + 1 is not atomic .

Reply:
Well: I tried to check that it did and I could not find clear confirmation that it was atomic, so I split it up into select/update. But I welcome a pointer to authoritative source that confirms that it is atomic!